### ISSUE TITLE
copyright and license

### ISSUE THREAD

**Issue by ghost** [wa0uwh/ERB-EspWebServer#1]
Hello, this project is really useful and interesting for the ESP8266 community but there is a copyright issue that prevent me to download it to test it. In your code, you indicate clearly that all rights are reserved but this prevent all the community to download it and install it to try to understand it, it also prevent any modifications in the attempt of study the code. It is clear that your effort is to make the source code public for everybody and this is great but you should protect your code with a different license, like a MIT license or BSD, there are many, Github also integrate tools to protect your code with a proper license, the license is important because it makes clear what we can do with your code. There are restrictive licenses that don't allow any commercial usage, put under the obligation of mentioning your name (attribute) but still allow the people to download your code and study it. Please adopt a proper opensource license to allow us to download it, actually is forbidden any uses. Thank you.

**Comment by wa0uwh** [comment]
Yes, I agree. But thus far I do not understand the proper copyright methods and have neglected the issue. I probable just need a standard template that would be less restrictive as I intended. My goal was to allow most people to use as they wish, but it would be nice if I participate in any profit that they may produce (but yet thinking that would not probably happen).

I will study the issue and see what others have done - this will be all new for me, pointers would be helpful.

Thanks,

Regards,
Eldon - WA0UWH

**Comment by ghost** [comment]
Hi Eldon,
sure you can protect your rights of do some business with the code along releasing it to the audience, you could give a look here: https://en.wikipedia.org/wiki/Copyleft
if you wish to get the attribution when somebody uses your code, you could get commoncreative license at: https://en.wikipedia.org/wiki/Creative_Commons_license and here https://creativecommons.org/choose/

I use mostly the MIT license or BSD 2, they are very permissive
thank you
